File: app/src/main/java/com/example/my_hospital_appointments/PatientMedicalReport.java
package com.example.my_hospital_appointments;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.widget.TextView;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.database.DataSnapshot;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;

public class PatientMedicalReport extends AppCompatActivity {
    @Override
    protected void onResume() {
        super.onResume();
        getReport(myUsersEmail);
    }

    TextView textViewTitle,textViewDoctorName,textViewResult,textViewConclusion;
    String myUsersEmail;
    DatabaseReference myDatabaseReports;
    @Override

    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_patient_medical_report);

        myDatabaseReports= FirebaseDatabase.getInstance().getReference("MedicalReport");
        Intent intent =getIntent();
        myUsersEmail=intent.getExtras().getString("usersEmail");

        textViewTitle=(TextView) findViewById(R.id.textViewTitleMedicalPATIENT);
        textViewDoctorName=(TextView) findViewById(R.id.textViewDoctorNameMedicalPATIENT);
        textViewResult=(TextView) findViewById(R.id.textViewtestResultMedicalPATIENT);
        textViewConclusion=(TextView) findViewById(R.id.textViewConclusionMedicalPATIENT);


    }

    private void getReport(String myKey)
    {
        String emailKey ="";
        int Counter=myKey.length();

        for(int a=0; a<Counter; a++)
        {
            if(myKey.charAt(a)=='@')
            {
                break;
            }
            else
            {
                emailKey=emailKey+myKey.charAt(a);

            }
        }

        emailKey=emailKey.trim();


       myDatabaseReports.child(emailKey).get().addOnCompleteListener(new OnCompleteListener<DataSnapshot>() {
           @Override
           public void onComplete(@NonNull Task<DataSnapshot> task) {

               if(task.isSuccessful())
               {
                   if(task.getResult().exists())
                   {
                       DataSnapshot thisDataSnapshot=task.getResult();
                       String title=String.valueOf(thisDataSnapshot.child("title").getValue());
                       String doctorName=String.valueOf(thisDataSnapshot.child("doctorName").getValue());
                       String testResult=String.valueOf(thisDataSnapshot.child("testResult").getValue());
                       String conclusion=String.valueOf(thisDataSnapshot.child("conclusion").getValue());

                       textViewDoctorName.setText(doctorName);
                       textViewTitle.setText(title);
                       textViewResult.setText(testResult);
                       textViewConclusion.setText(conclusion);

                   }
                   else
                   {
                       String notYet="NOT AVAILABLE";
                       textViewDoctorName.setText(notYet);
                       textViewTitle.setText(notYet);
                       textViewResult.setText(notYet);
                       textViewConclusion.setText(notYet);
                   }
               }
           }
       });
    }
}

File: app/src/main/java/com/example/my_hospital_appointments/PatientMedicalReport.kt
package com.example.my_hospital_appointments

import android.os.Bundle
import android.view.View
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.database.DatabaseReference
import com.google.firebase.database.FirebaseDatabase

class PatientMedicalReport : AppCompatActivity() {
    override fun onResume() {
        super.onResume()
        getReport(myUsersEmail)
    }

    var textViewTitle: TextView? = null
    var textViewDoctorName: TextView? = null
    var textViewResult: TextView? = null
    var textViewConclusion: TextView? = null
    var myUsersEmail: String? = null
    var myDatabaseReports: DatabaseReference? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_patient_medical_report)
        myDatabaseReports = FirebaseDatabase.getInstance().getReference("MedicalReport")
        val intent = intent
        myUsersEmail = intent.extras!!.getString("usersEmail")
        textViewTitle = findViewById<View>(R.id.textViewTitleMedicalPATIENT) as TextView
        textViewDoctorName = findViewById<View>(R.id.textViewDoctorNameMedicalPATIENT) as TextView
        textViewResult = findViewById<View>(R.id.textViewtestResultMedicalPATIENT) as TextView
        textViewConclusion = findViewById<View>(R.id.textViewConclusionMedicalPATIENT) as TextView
    }

    private fun getReport(myKey: String?) {
        var emailKey = ""
        val Counter = myKey!!.length
        for (a in 0 until Counter) {
            emailKey = if (myKey[a] == '@') {
                break
            } else {
                emailKey + myKey[a]
            }
        }
        emailKey = emailKey.trim { it <= ' ' }
        myDatabaseReports!!.child(emailKey).get().addOnCompleteListener { task ->
            if (task.isSuccessful) {
                if (task.result.exists()) {
                    val thisDataSnapshot = task.result
                    val title = thisDataSnapshot.child("title").value.toString()
                    val doctorName = thisDataSnapshot.child("doctorName").value.toString()
                    val testResult = thisDataSnapshot.child("testResult").value.toString()
                    val conclusion = thisDataSnapshot.child("conclusion").value.toString()
                    textViewDoctorName!!.text = doctorName
                    textViewTitle!!.text = title
                    textViewResult!!.text = testResult
                    textViewConclusion!!.text = conclusion
                } else {
                    val notYet = "NOT AVAILABLE"
                    textViewDoctorName!!.text = notYet
                    textViewTitle!!.text = notYet
                    textViewResult!!.text = notYet
                    textViewConclusion!!.text = notYet
                }
            }
        }
    }
}